autoload/rustfmt.vim

```diff
@@ -146,8 +146,8 @@ function! s:RunRustfmt(command, tmpname, fail_silently)
             let l:content = readfile(a:tmpname)
         endif
 
-        call writefile(l:content, expand('%'))
-        silent edit!
+        normal! ggdG
+        call setline(1, l:content)
         let &syntax = &syntax
```

Comment:
The `let &syntax = &syntax` is also not necessary anymore likely (if it ever was, but maybe due to the `edit` that was used previously).

TLDR of the investigation: the problem is that `expand('%')` here might be relative, but it was `:lcd`ed before.
An alternative might be to use the full buffer name (save it before `cd`ing), but I think just setting the lines is better than writing from outside.

Reply:
Actually `expand('%')` should be the relative path after `:lcd` - so there is some other reason for this (maybe a bug in Vim/Neovim even (NVIM v0.3.1 used by the reporter)).

Anyway, the `cd`ing back should be done in a `try … finally` block, because otherwise it might not change it back.